### PR TITLE
pkg: make `seed` in `ImageType.Manifest()` random by default

### PR DESCRIPTION
**Pull request by mvo5**
This commit changes the API of `ImageType.Manifest()` to take a pointer to the seed value. If no seed value is given it will automatically generate a random one.

This avoid the pitfall that the consumer needs to know about properly seeding the manifests. The default use-case is to generate non-fixed seed manifests so this commit makes it easier.

[edit: this is one possible way to avoid https://github.com/osbuild/image-builder-cli/pull/6#discussion_r1882807289]

### CODE REVIEW

**Review by ondrejbudai — COMMENTED** [review]
I like that it's safe by default.

File: pkg/distro/distro.go [hunk 1]
```diff
@@ -120,7 +122,9 @@ type ImageType interface {
 	// specified in the given blueprint; it also returns any warnings (e.g.
 	// deprecation notices) generated by the manifest.
 	// The packageSpecSets must be labelled in the same way as the originating PackageSets.
-	Manifest(bp *blueprint.Blueprint, options ImageOptions, repos []rpmmd.RepoConfig, seed int64) (*manifest.Manifest, []string, error)
+	// A custom seed for the rng can be specified, if nil the seed will
+	// be random.
```
Comment (achilleas-k) [inline pkg/distro/distro.go:126]: Maybe we can be more accurate and informative here and say that it will be derived from the current time.

Comment (mvo5) [reply pkg/distro/distro.go:126]: Thanks! I'm happy to do that, I was thinking of using crypto.Rand to seed the generator but maybe I'm overthinking it. Happy to either document or change.

**Review by thozza — APPROVED** [review]
Thanks, this looks OK to me.

**Review by achilleas-k — APPROVED** [review]

File: pkg/distro/distro.go (hunk 1 above)
Comment (achilleas-k) [reply pkg/distro/distro.go:126]: Not important.  Can do later or not at all.

**Comment by mvo5** [comment]
I prepared the followups to fix the consumers

**Pull request merged**